src/LayoutManager/ActiveSelectionLayoutManager.spec.ts

```diff
@@ -0,0 +1,89 @@
+import type { TModificationEvents } from '../EventTypeDefs';
+import { ActiveSelection } from '../shapes/ActiveSelection';
+import { Group } from '../shapes/Group';
+import { FabricObject } from '../shapes/Object/FabricObject';
+import { ActiveSelectionLayoutManager } from './ActiveSelectionLayoutManager';
+
+describe('ActiveSelectionLayoutManager', () => {
+  describe('onBeforeLayout', () => {
+    describe('triggers', () => {
+      const triggers: ('modified' | TModificationEvents | 'changed')[] = [
+        'modified',
+        'moving',
+        'resizing',
+        'rotating',
+        'scaling',
+        'skewing',
+        'changed',
+        'modifyPoly',
+      ];
+
+      it('should subscribe activeSelection that contains object', () => {
+        const manager = new ActiveSelectionLayoutManager();
+        const object = new FabricObject();
+        const group = new Group([object], {
+          interactive: true,
+          subTargetCheck: true,
+        });
+        const as = new ActiveSelection([object], { layoutManager: manager });
+        const objectOn = jest.spyOn(object, 'on');
+        const objectOff = jest.spyOn(object, 'off');
+        const asOn = jest.spyOn(as, 'on');
+        manager.subscribeTargets({
+          targets: [object],
+          target: as,
+        });
+        expect(objectOn).not.toHaveBeenCalled();
+        expect(objectOff).not.toHaveBeenCalled();
+        expect(asOn).toHaveBeenCalledTimes(triggers.length);
+        expect(objectOff).not.toHaveBeenCalled();
+      });
```

Comment:
This test is missing assertion checking double registration of objects with the same parent not registering the AS twice

Reply:
I m not sure with this new code structure they will register once only